### PR TITLE
ci: add simple rust workflow to build and run tests

### PR DESCRIPTION
**Pull request by MatejVukosav**
- Use caching for dependencies and build artifacts to avoid recompiling unchanged code.
- Split jobs into parallel tasks to optimize CI time.
- Leverage incremental builds during development.
- Consider sccache to cache compiled artifacts across different runs.

### CODE REVIEW

**Review by xilosada — APPROVED** [review]

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install sccache
+      run: |
+        wget https://github.com/mozilla/sccache/releases/download/v0.5.3/sccache-v0.5.3-x86_64-unknown-linux-musl.tar.gz
+        tar -xzf sccache-v0.5.3-x86_64-unknown-linux-musl.tar.gz
+        sudo mv sccache-v0.5.3-x86_64-unknown-linux-musl/sccache /usr/local/bin/
+        rm -rf sccache-v0.5.3-x86_64-unknown-linux-musl*
```
Comment (miraclx) [inline .github/workflows/rust.yml]: this is [a year old](https://github.com/mozilla/sccache/releases/tag/v0.5.3), why aren't we using the github action?

https://github.com/marketplace/actions/sccache-action

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    - name: Cache Cargo and sccache
+      uses: actions/cache@v4
+      with:
+        path: |
+          ~/.cargo/registry
+          ~/.cargo/git
+          target
+          ~/.cache/sccache
```
Comment (miraclx) [inline .github/workflows/rust.yml]: we should be using https://github.com/Swatinem/rust-cache here

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -14,34 +14,28 @@ jobs:
...
+      - name: Cache Cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-${{ hashFiles('rust-toolchain.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
```
Comment (miraclx) [inline .github/workflows/rust.yml]: this is now handled by `rust-cache`

Comment (MatejVukosav) [reply .github/workflows/rust.yml]: Wasn't that outcome of comment before?

Comment (miraclx) [reply .github/workflows/rust.yml]: rust-cache will cache, and [knows the right things to include in the cache](https://github.com/Swatinem/rust-cache#cache-details), we don't need to duplicate it here

**Review by miraclx — APPROVED** [review]

**Pull request merged**